packages/shell-evaluator/src/shell-evaluator.ts

```diff
@@ -64,10 +67,13 @@ class ShellEvaluator {
         this.saveState();
         const rewrittenInput = this.internalState.asyncWriter.process(input);
 
-        this.internalState.messageBus.emit(
-          'mongosh:rewritten-async-input',
-          { original: input.trim(), rewritten: rewrittenInput.trim() }
-        );
+        const hiddenCommands = RegExp(HIDDEN_COMMANDS, 'g');
+        if (!hiddenCommands.test(input) && !hiddenCommands.test(rewrittenInput)) {
```

Comment:
Could the input here contain more commands than just the hidden commands? Are those also dropped?

Reply:
If the input/output does not include any hidden commands then it will emit the log message for those commands, so they are not dropped. We want to not log for the hidden commands, which contain sensitive information like passwords.